src/sbvr-api/sbvr-utils.ts

```diff
@@ -1205,17 +1287,22 @@ const runODataRequest = (req: Express.Request, vocabulary: string) => {
 			await runHooks('PREPARSE', reqHooks, { req, tx: req.tx });
 			let requests: uriParser.UnparsedRequest[];
 			// Check if it is a single request or a batch
-			if (req.batch != null && req.batch.length > 0) {
-				requests = req.batch;
+			if (req.url === `/${vocabulary}/$batch`) {
+				requests = req.batch!;
 			} else {
 				const { method, url, body } = req;
-				requests = [{ method, url, data: body }];
+				requests = [{ method, url, body }];
 			}
 
 			const prepareRequest = async (
 				parsedRequest: uriParser.ParsedODataRequest &
 					Partial<Pick<uriParser.ODataRequest, 'engine' | 'translateVersions'>>,
 			): Promise<uriParser.ODataRequest> => {
+				if (models[parsedRequest.vocabulary] == null) {
+					throw new BadRequestError(
+						'Unknown vocabulary: ' + parsedRequest.vocabulary,
+					);
+				}
```

Comment:
I think this can be merged in a separate PR to shrink this one a bit

Reply:
I assume you mean as a PR preceding this one? See: https://github.com/balena-io/pinejs/pull/678